HS256 inline support inconsistent

Following up to https://github.com/quarkusio/quarkus/discussions/42550 and https://github.com/smallrye/smallrye-jwt/issues/814 . Consolidating all the test cases here:

Defining the variables...
- `secretKey.jwk` contains a `k` that is BASE64 secret key, as follows:

```
{
  "kty": "oct",
  "kid": "Sb6ml8Bc7sTDQw/",
  "k": "cDEzUGZEZEMzam1KUHBNMFFUNWR6L3pwdHRSRWJJelJ1ZE56MzRJbEhEZUtrVy9LZjJkVGhibVJaUVhQU3IveDdYU3BvN2FjSmNxbmdhbmFRejdSQTFnPT0=",
  "alg": "HS256"
}
```

- defining a `jwk_base64` that is the entire  `secretKey.jwk` file converted into a BASE64URL string, that is:
```
ew0KICAia3R5IjogIm9jdCIsDQogICJraWQiOiAiU2I2bWw4QmM3c1REUXcvIiwNCiAgImsiOiAiY0RFelVHWkVaRU16YW0xS1VIQk5NRkZVTldSNkwzcHdkSFJTUldKSmVsSjFaRTU2TXpSSmJFaEVaVXRyVnk5TFpqSmtWR2hpYlZKYVVWaFFVM0l2ZURkWVUzQnZOMkZqU21OeGJtZGhibUZSZWpkU1FURm5QVDA9IiwNCiAgImFsZyI6ICJIUzI1NiINCn0
```

- there also exist a file `secretKey_url.jwk` that contains only the content of `jwk_base64`

---

Now for the 2 successful use cases: comes from using the JWKs as files in either JSON or BASE64URL format:

```
smallrye.jwt.verify.key-format = JWK
smallrye.jwt.verify.key.location = secretKey.jwk
smallrye.jwt.verify.algorithm = HS256
```

and 

```
smallrye.jwt.verify.key-format = JWK_BASE64URL
smallrye.jwt.verify.key.location = secretKey_url.jwk
smallrye.jwt.verify.algorithm = HS256
```
---

Unsuccessful cases include different ways of using the symmetric key inline:
```
smallrye.jwt.verify.key-format = JWK_BASE64URL
mp.jwt.verify.publickey = ew0KICAia3R5IjogIm9jdCIsDQogICJraWQiOiAiU2I2bWw4QmM3c1REUXcvIiwNCiAgImsiOiAiY0RFelVHWkVaRU16YW0xS1VIQk5NRkZVTldSNkwzcHdkSFJTUldKSmVsSjFaRTU2TXpSSmJFaEVaVXRyVnk5TFpqSmtWR2hpYlZKYVVWaFFVM0l2ZURkWVUzQnZOMkZqU21OeGJtZGhibUZSZWpkU1FURm5QVDA9IiwNCiAgImFsZyI6ICJIUzI1NiINCn0
smallrye.jwt.verify.algorithm = HS256
```
leading to a `error id: e4239d1a-0464-436e-b22b-e8c82e11e432-5: jakarta.enterprise.inject.spi.DeploymentException: SRJWT02000: HMAC verification algorithms are not supported when the 'mp.jwt.verify.publickey.location' property is set, use 'smallrye.jwt.verify.key.location' instead` error, which is odd because I don't have `smallrye.jwt.verify.key.location` set in this case

---

The following cases all lead to the same error. If I use the BASE64 JWT:
```
mp.jwt.verify.algorithm=HS256
mp.jwt.verify.publickey = ew0KICAia3R5IjogIm9jdCIsDQogICJraWQiOiAiU2I2bWw4QmM3c1REUXcvIiwNCiAgImsiOiAiY0RFelVHWkVaRU16YW0xS1VIQk5NRkZVTldSNkwzcHdkSFJTUldKSmVsSjFaRTU2TXpSSmJFaEVaVXRyVnk5TFpqSmtWR2hpYlZKYVVWaFFVM0l2ZURkWVUzQnZOMkZqU21OeGJtZGhibUZSZWpkU1FURm5QVDA9IiwNCiAgImFsZyI6ICJIUzI1NiINCn0
```
Or using `k`:
```
mp.jwt.verify.algorithm=HS256
mp.jwt.verify.publickey = cDEzUGZEZEMzam1KUHBNMFFUNWR6L3pwdHRSRWJJelJ1ZE56MzRJbEhEZUtrVy9LZjJkVGhibVJaUVhQU3IveDdYU3BvN2FjSmNxbmdhbmFRejdSQTFnPT0=
```
Or using the JWT file
```
mp.jwt.verify.algorithm = HS256
mp.jwt.verify.publickey.location = secretKey.jwk
```
Or using the BASE file
```
mp.jwt.verify.algorithm = HS256
mp.jwt.verify.publickey.location = secretKey_url.jwk
```

all leads to a ` Authentication failed: io.smallrye.jwt.auth.principal.ParseException: SRJWT07000: Failed to verify a token` and `org.jose4j.lang.UnresolvableKeyException: SRJWT07004: Failed to load a key from the key content while resolving` error. 


---

As a sidenote, I pulled [64becff](https://github.com/smallrye/smallrye-jwt/commit/64becff652f3745ede6fcb5d6d4fee4e167de725) to test https://github.com/smallrye/smallrye-jwt/issues/814 and tried to build it locally, but wasn't able to build it. I get this error during `mvn clean install -X`:

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for SmallRye: MicroProfile JWT Parent 4.5.4-SNAPSHOT:
[INFO]
[INFO] SmallRye: MicroProfile JWT Parent .................. SUCCESS [  1.493 s]
[INFO] SmallRye: MicroProfile JWT - Implementation Parent . SUCCESS [  3.527 s]
[INFO] SmallRye: MicroProfile JWT Implementation Common ... SUCCESS [  6.639 s]
[INFO] SmallRye: MicroProfile JWT Build Implementation .... SUCCESS [  8.044 s]
[INFO] SmallRye: MicroProfile JWT Implementation .......... SUCCESS [ 17.910 s]
[INFO] SmallRye: MicroProfile JWT JAX-RS Implementation ... SUCCESS [  3.501 s]
[INFO] SmallRye: MicroProfile JWT HTTP Mechanism Implementation SUCCESS [  3.238 s]
[INFO] SmallRye: MicroProfile JWT CDI Extension Implementation SUCCESS [  2.938 s]
[INFO] SmallRye: MicroProfile JWT - Testsuite Parent ...... SUCCESS [  0.034 s]
[INFO] SmallRye: MicroProfile JWT Basic Testsuite ......... SUCCESS [  5.496 s]
[INFO] SmallRye: MicroProfile JWT TCK ..................... FAILURE [  4.378 s]
[INFO] Empty Release Project to Avoid Maven Bug ........... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------

ERROR] Please refer to C:\github\smallrye-jwt\testsuite\tck\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] There was an error in the forked process
[ERROR] Suite file C:\github\smallrye-jwt\testsuite\tck\target\tck-suite\suites\tck-base-suite.xml is not a valid file
[ERROR] org.apache.maven.surefire.api.testset.TestSetFailedException: Suite file C:\github\smallrye-jwt\testsuite\tck\target\tck-suite\suites\tck-base-suite.xml is not a valid file
[ERROR]         at org.apache.maven.surefire.testng.TestNGXmlTestSuite.locateTestSets(TestNGXmlTestSuite.java:89)
[ERROR]         at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:112)
[ERROR]         at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
[ERROR]         at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
[ERROR]         at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
[ERROR]         at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
[ERROR]
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: There was an error in the forked process
[ERROR] Suite file C:\github\smallrye-jwt\testsuite\tck\target\tck-suite\suites\tck-base-suite.xml is not a valid file
[ERROR] org.apache.maven.surefire.api.testset.TestSetFailedException: Suite file C:\github\smallrye-jwt\testsuite\tck\target\tck-suite\suites\tck-base-suite.xml is not a valid file
[ERROR]         at org.apache.maven.surefire.testng.TestNGXmlTestSuite.locateTestSets(TestNGXmlTestSuite.java:89)
[ERROR]         at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:112)
[ERROR]         at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
[ERROR]         at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
[ERROR]         at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
[ERROR]         at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
[ERROR]
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:628)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:285)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:250)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1224)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1073)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:907)
[ERROR]         at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:328)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:316)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:174)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:75)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:162)
[ERROR]         at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:159)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
[ERROR]         at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
[ERROR]         at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
[ERROR]         at org.apache.maven.cli.MavenCli.execute(MavenCli.java:903)
[ERROR]         at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:280)
[ERROR]         at org.apache.maven.cli.MavenCli.main(MavenCli.java:203)
[ERROR]         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:255)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:201)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:361)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:314)
```

Trying to use this file: https://github.com/eclipse/microprofile-jwt-auth/blob/main/tck/src/test/resources/suites/tck-base-suite.xml and saving it at a few different locations under `testsuites\tck-suite\suites\` doesn't seem to field anything helpful.

--- a/implementation/jwt-auth/src/main/java/io/smallrye/jwt/config/JWTAuthContextInfoProvider.java
+++ b/implementation/jwt-auth/src/main/java/io/smallrye/jwt/config/JWTAuthContextInfoProvider.java
@@ -841,7 +841,7 @@ public class JWTAuthContextInfoProvider {
             resolvedAlgorithm = mpJwtPublicKeyAlgorithm;
         } else if (signatureAlgorithm.isPresent()) {
             if (signatureAlgorithm.get().getAlgorithm().startsWith("HS")) {
-                if (resolvedVerifyKeyLocation == mpJwtLocation) {
+                if (!NONE.equals(resolvedVerifyKeyLocation) && resolvedVerifyKeyLocation == mpJwtLocation) {
                     throw ConfigMessages.msg.hmacNotSupported();
                 }
             } else {

